packages/angular/src/flyout-menu/flyout-menu.component.ts

```diff
@@ -0,0 +1,89 @@
+import { Component, Input, OnInit, ViewEncapsulation } from '@angular/core';
+import { IconService } from 'carbon-components-angular';
+import { Filter16 } from '@carbon/icons';
+
+/**
+ * The Filter menu component encapsulates the OverFlowMenu directive, and the flyout iconography
+ * into one convienent component
+ *
+ * [See demo](../../?path=/story/components-flyout-menu--basic)
+ *
+ * html:
+ * ```
+ * <ai-flyout-menu>
+ *	options
+ * </ai-flyout-menu>
+ * ```
+ *
+ * <example-url>../../iframe.html?id=components-flyout-menu--basic</example-url>
+ */
+@Component({
+  selector: 'ai-flyout-menu',
+  template: `
+    <ng-template #templateRef let-tooltip="tooltip">
+      <div class="bx--tooltip__content">
+        <div class="iot--flyout-menu--content">
+          <ng-content></ng-content>
+        </div>
+        <div class="iot--flyout-menu__bottom-container">
+          <ng-content select="[cancelButton]"></ng-content>
+          <ng-content select="[applyButton]"></ng-content>
```

Comment:
If we always have cancel/apply it may be easier to just include them here and have a `@Output` for `cancel` and `apply`

Reply:
Or make a footer/button container component that can include them and fully pass of the responsibility to the app ... How would the app signal that the dialog should close in this case? Or is that something that makes more sense for the component to control anyway?

Reply:
They are optional so in addition to events, we'd also need inputs to enable/disable them.
Selectors allow full flexibility, so that's what I went with instead.

As the designs evolve, we might want to revisit that.

Reply:
Sorry, I guess I wasn't super clear, how about something like - 

```html
<ai-flyout-footer>
  <button ibmButton="secondary">Cancel</button>
  <button ibmButton="primary">Apply</button>
</ai-flyout-footer>
```

That way we don't have to do any extra work (the styles should handle laying out the buttons) and the consumer gets the same flexibility in specifying the button control without having to add random attributes to the buttons (plus they can add N extra buttons, which may be useful?)

If we wanted to be really helpful we could _also_ add a `ai-flyout-footer-buttons` component that wraps up the footer + the default set of buttons, but then we run into the points you mentioned (events + inputs + g11n concerns)